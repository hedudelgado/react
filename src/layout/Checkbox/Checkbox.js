import React from 'react';
import { bool, object, string } from 'prop-types';
import classnames from 'classnames';

const Checkbox = ({
  label,
  field,
  id,
  sessioncamhide,
  ...props
}) => {
  const classes = classnames({
    sessioncamhidetext: sessioncamhide,
  });
  return (
    <div className="wb-form-item wb-checkbox wb-checkbox__white wb-form-item--clean mb3">
      <input
        type="checkbox"
        id={id}
        className={classes}
        {...field}
        {...props}
      />
      <label htmlFor={id}>{label}</label>
    </div>
  );
};

Checkbox.propTypes = {
  label: string.isRequired,
  field: object.isRequired,
  id: string.isRequired,
  /**
    Sessioncam field masking.
  */
  sessioncamhide: bool,
};

export default Checkbox;
