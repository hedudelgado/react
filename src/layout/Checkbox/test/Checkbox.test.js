import React from 'react';
import { shallow } from 'enzyme';

import Checkbox from '../Checkbox';

describe('Checkbox Component', () => {
  const props = {
    label: 'test',
    field: {},
    id: 'test',
  };

  describe('Renders Checkbox Element', () => {
    it('should render the component', () => {
      const wrapper = shallow(<Checkbox {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
