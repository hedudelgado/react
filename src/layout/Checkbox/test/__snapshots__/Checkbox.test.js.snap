// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Component Renders Checkbox Element should render the component 1`] = `
<div
  className="wb-form-item wb-checkbox wb-checkbox__white wb-form-item--clean mb3"
>
  <input
    className=""
    id="test"
    type="checkbox"
  />
  <label
    htmlFor="test"
  >
    test
  </label>
</div>
`;
